scripts/openvswitch-config-update

```diff
@@ -0,0 +1,300 @@
+            m = re.match('.*hwaddr="([0-9a-fA-F:].*)".*', oc)
+            if m and m.group(1) in pool_mgmt_macs.keys():
+                mgmt_bridge = bridge
+                host_mgmt_mac = m.group(1)
+                host_mgmt_device = pool_mgmt_macs[host_mgmt_mac]
+
+    if (host_mgmt_mac is not None and mgmt_bridge is not None and
+            host_mgmt_device is not None):
+        tp = 'idle_timeout=0,priority=0'
+        port = vswitchCfgQuery(['get', 'interface', host_mgmt_device,
+                                'ofport'])
+
+        addFlow(mgmt_bridge, '%s,in_port=%s,arp,nw_proto=1,actions=local' %
+                (tp, port))
+        addFlow(mgmt_bridge, '%s,in_port=local,arp,dl_src=%s,actions=%s' %
+                (tp, host_mgmt_mac, port))
+        addFlow(mgmt_bridge, '%s,in_port=%s,dl_dst=%s,actions=local' %
+                (tp, port, host_mgmt_mac))
+        addFlow(mgmt_bridge, '%s,in_port=local,dl_src=%s,actions=%s' %
+                (tp, host_mgmt_mac, port))
+
+    if dib_changed:
+        ret_str += 'Updated in-band management.  '
+    if fail_mode_changed:
+        ret_str += 'Updated fail_mode.  '
+
+    if ret_str != '':
+        return ret_str
+    else:
+        return 'No change to configuration'
+
+
+def vswitchCurrentControllers():
+    controllers = vswitchCfgQuery(['get-manager'])
+    return controllers.split('\n')
+
+
+def removeControllerCfg():
+    vswitchCfgMod(['--', 'del-manager',
+                   '--', 'del-ssl'])
+
+
+def setControllerCfg(bootstrap, controller):
+    # /etc/xensource/xapi-ssl.pem is mentioned twice below because it
+    # contains both the private key and the certificate.
+    if bootstrap:
+        # For active ssl, do not need to turn on TCP port 6640 on host.
+        cmd = [iptablectl, 'close', '6640']
+        subprocess.call(cmd)
+        vswitchCfgMod(['--', 'del-manager',
+                       '--', 'del-ssl',
+                       '--', '--bootstrap', 'set-ssl',
+                       '/etc/xensource/xapi-ssl.pem',
+                       '/etc/xensource/xapi-ssl.pem',
+                       bootstrap_cacert,
+                       '--', 'set-manager', controller])
+    else:
+        # For passive ssl, only TCP port 6640 is meaningful.
+        # Turn on the port of the host.
+        cmd = [iptablectl, 'open', '6640']
+        subprocess.call(cmd)
+        vswitchCfgMod(['--', 'del-manager',
+                       '--', 'del-ssl',
+                       '--', 'set-ssl',
+                       '/etc/xensource/xapi-ssl.pem',
+                       '/etc/xensource/xapi-ssl.pem',
+                       preinstalled_cacert,
+                       '--', 'set-manager', controller])
+
+
+def composeController(protocol, controllerIP, tcpport):
+    bootstrap = False
+    controller = ''
+    if protocol == 'ssl':
+        bootstrap = True
+        controller = protocol + ':' + controllerIP + ':' + tcpport
+    elif protocol == 'pssl':
+        if controllerIP == '':
```

Comment:
I don't think `controllerIP` is what we want here, because for pssl, the IP is the _local_ IP. Leaving it empty means that the OVS will listen on all of the host's IP addresses, while specifying an IP causes it to listen on just that IP.